Search results / Add template to display related record as list instead of dropdown

fxprunayre opened the pull request:
Currently related records are displayed in a dropdown
![image](https://user-images.githubusercontent.com/1701393/60574339-bb455e80-9d79-11e9-8625-f8b53aa60c27.png)

In grid mode, it is probably good as we don't want to have variable height depending on number of related. When using row mode, this template is useful:

![image](https://user-images.githubusercontent.com/1701393/60574303-a8328e80-9d79-11e9-9c3b-2bca020f316e.png)

MichelGabriel commented:
Is this going to be the default option for the 'row' mode? Or can you choose to use it?

And indeed, in 'grid' a variable height is not possible if you don't want to mess up the whole grid.

fxprunayre commented:
> Is this going to be the default option for the 'row' mode? Or can you choose to use it?

Yep maybe I could add this and we can have by default the 2 mode (with grid mode by default) ?

fxprunayre commented:
@MichelGabriel, I added it to the list mode (not activated by default).

![image](https://user-images.githubusercontent.com/1701393/60791124-469c6680-a163-11e9-8111-8aba0bd784d8.png)

And also added a more/less button (thanks @davinciagf) in case of large list

![image](https://user-images.githubusercontent.com/1701393/60791186-63d13500-a163-11e9-8d05-c4dd193bb32b.png)

So if you have time to review thanks ?


Then before activating the list mode, maybe you can review it a bit and make another PR - there is some grey borders which might need improvements.

![image](https://user-images.githubusercontent.com/1701393/60791280-95e29700-a163-11e9-900f-c6f05f622084.png)

MichelGabriel commented:
The grey borders look like thumbnails not there. I will have a look

MichelGabriel commented:
I fixed some of these issues in a PR I made some time ago: https://github.com/geonetwork/core-geonetwork/pull/3874

fxprunayre commented:
> I fixed some of these issues in a PR I made some time ago: #3874

Ok I merged it and fix conflict here. If this one looks good to you, then approve and we can merge.

MichelGabriel commented:
Some small remarks:
- it would be nice to have a title of some sort
- I like it when the first level of the list is not indented so much (on the same line as the paragraph above it
- in the grid view the icons (at least for me) are not aligned:
![gn-search-align-buttons](https://user-images.githubusercontent.com/19608667/60971501-98401f00-a324-11e9-8a13-98405aa41711.png)

The rest looks great, thanks

fxprunayre commented:
> * it would be nice to have a title of some sort

![image](https://user-images.githubusercontent.com/1701393/61027110-a2f6c480-a3b5-11e9-8b66-094cd0cd0f73.png)


> I like it when the first level of the list is not indented so much (on the same line as the paragraph above it

![image](https://user-images.githubusercontent.com/1701393/61027105-9ffbd400-a3b5-11e9-9717-44e522b59a7d.png)


> in the grid view the icons (at least for me) are not aligned:

This is new and fixed now
